src/Umbraco.Community.Contentment/Services/Implement/ContentmentContextAccessor.cs

```diff
@@ -0,0 +1,85 @@
+﻿#if NET472
+using Umbraco.Core.Models.PublishedContent;
+using Umbraco.Web;
+#else
+using Umbraco.Cms.Core.Models.PublishedContent;
+using Umbraco.Cms.Core.Web;
+using Umbraco.Extensions;
+#endif
+
+namespace Umbraco.Community.Contentment.Services.Implement
+{
+    public class ContentmentContextAccessor : IContentmentContextAccessor
+    {
+        private readonly IUmbracoContextAccessor _umbracoContextAccessor;
+
+#if NET472
+        public ContentmentContextAccessor(
+            IUmbracoContextAccessor umbracoContextAccessor)
+        {
+            _umbracoContextAccessor = umbracoContextAccessor;
+
+        }
+#else
+        private readonly IRequestAccessor _requestAccessor;
+
+        public ContentmentContextAccessor(
+            IRequestAccessor requestAccessor,
+            IUmbracoContextAccessor umbracoContextAccessor)
+        {
+            _requestAccessor = requestAccessor;
+            _umbracoContextAccessor = umbracoContextAccessor;
+
+        }
+#endif
+
+
+        public int? GetCurrentContentId(out bool isParent)
+        {
+            var umbracoContext = _umbracoContextAccessor.GetRequiredUmbracoContext();
```

Comment:
I'd use `_umbracoContextAccessor.TryGetUmbracoContext(out var umbracoContext)` here. Only really to cover myself for any unexpected cases where the `UmbracoContext` isn't there. (I can't think of a scenario where it wouldn't be there, but who knows! 🤷 😅)